### PR TITLE
Add descriptions to all the `rdctl set` options

### LINKED ISSUE #4070: Add option descriptions for all new options in command-api.yaml
We can add a usage message via the `x-rd-usage` key that will be displayed by `rdctl set --help`, but most options are missing the description:

```
$ rdctl set --help
Update selected fields in the Rancher Desktop UI and restart the backend.

Usage:
  rdctl set [flags]

Flags:
      --application.admin-access
      --application.auto-start
      --application.debug
      --application.hide-notification-icon
      --application.path-management-strategy string                     (Allowed values: [manual, rcfiles])
      --application.start-in-background
      --application.telemetry.enabled
      --application.updater.enabled
      --application.window.quit-on-close
      --container-engine.allowed-images.enabled
      --container-engine.allowed-images.locked
      --container-engine.name string                                    Set engine to containerd or moby (aka docker). (Allowed values: [containerd, docker, moby])
      --diagnostics.show-muted
      --experimental.virtual-machine.mount.9p.cache-mode string         (Allowed values: [none, loose, fscache, mmap])
      --experimental.virtual-machine.mount.9p.msize-in-kb int
      --experimental.virtual-machine.mount.9p.protocol-version string   (Allowed values: [9p2000, 9p2000.u, 9p2000.L])
      --experimental.virtual-machine.mount.9p.security-model string     (Allowed values: [passthrough, mapped-xattr, mapped-file, none])
      --experimental.virtual-machine.mount.type string                  (Allowed values: [reverse-sshfs, 9p, virtiofs])
      --experimental.virtual-machine.socket-vmnet
      --experimental.virtual-machine.type string                        (Allowed values: [qemu, vz])
      --experimental.virtual-machine.use-rosetta
  -h, --help                                                            help for set
      --images.namespace string
      --images.show-all
      --kubernetes.enabled                                              Control whether kubernetes runs in the backend.
      --kubernetes.options.flannel                                      Control whether flannel is enabled. Use to disable flannel so you can install your own CNI.
      --kubernetes.options.traefik
      --kubernetes.port int
      --kubernetes.version string                                       Choose which version of kubernetes to run.
      --port-forwarding.include-kubernetes-services
      --virtual-machine.memory-in-gb int
      --virtual-machine.number-cpus int
```

### PR DESCRIPTION
**Pull request by jandubois**
Also includes some cleanup of the YAML formatting.

```console
Flags:
      --application.admin-access                                        Enable privileged operations
      --application.auto-start                                          Start app when logging in.
      --application.debug                                               Generate more verbose logging.
      --application.hide-notification-icon                              Don't show notification icon.
      --application.path-management-strategy string                     Update PATH to include ~/.rd/bin. (Allowed values: [manual, rcfiles])
      --application.start-in-background                                 Start app without window.
      --application.telemetry.enabled                                   Allow collection of anonymous statistics.
      --application.updater.enabled                                     Automatically update to the latest release.
      --application.window.quit-on-close                                Terminate app when the main window is closed.
      --container-engine.allowed-images.enabled                         Only allow images to be pulled that match the allowed patterns.
      --container-engine.name string                                    Set engine. (Allowed values: [containerd, docker, moby])
      --diagnostics.show-muted                                          Unhide muted diagnostics.
      --experimental.virtual-machine.mount.9p.cache-mode string         (Allowed values: [none, loose, fscache, mmap])
      --experimental.virtual-machine.mount.9p.msize-in-kb int           Maximum packet size.
      --experimental.virtual-machine.mount.9p.protocol-version string   (Allowed values: [9p2000, 9p2000.u, 9p2000.L])
      --experimental.virtual-machine.mount.9p.security-model string     (Allowed values: [passthrough, mapped-xattr, mapped-file, none])
      --experimental.virtual-machine.mount.type string                  Choose how directories are shared. (Allowed values: [reverse-sshfs, 9p])
      --experimental.virtual-machine.socket-vmnet                       Use socket-vmnet instead of vde-vmnet.
  -h, --help                                                            help for set
      --images.namespace string                                         Select only images from this namespace (containerd only).
      --images.show-all                                                 Show system images on Images page.
      --kubernetes.enabled                                              Control whether kubernetes is started.
      --kubernetes.options.flannel                                      Use flannel networking. Disable to install your own CNI.
      --kubernetes.options.traefik                                      Install and run traefik.
      --kubernetes.port int                                             API server port.
      --kubernetes.version string                                       Choose which version of kubernetes to run.
      --port-forwarding.include-kubernetes-services                     Show Kubernetes system services on Port Forwarding page.
      --virtual-machine.memory-in-gb int                                Reserved RAM size.
      --virtual-machine.number-cpus int                                 Reserved number of CPUs.

Global Flags:
      --config-path string   config file (default /Users/jan/Library/Application Support/rancher-desktop/rd-engine.json)
      --host string          default is localhost; most useful for WSL
      --password string      overrides the password setting in the config file
      --port string          overrides the port setting in the config file
      --user string          overrides the user setting in the config file
```

Style is a bit different between command flags descriptions (start with uppercase letter; trailing dot), and the global flags (lowercase, no trailing dot), but I'll leave that for another day / another PR.

Fixes #4070

### CODE REVIEW

File: pkg/rancher-desktop/assets/specs/command-api.yaml [hunk 1]
```diff
@@ -238,50 +238,63 @@ components:
           properties:
             adminAccess:
               type: boolean
-              x-rd-platforms: ['darwin', 'linux'] # Only in the specified platforms
+              x-rd-usage: Enable privileged operations
```
Comment (ericpromislow) [inline pkg/rancher-desktop/assets/specs/command-api.yaml:241]: Missing period at the end, but I think we would do better by dropping the final period for single-sentence descriptions. Makes it less cluttered. And all descriptions should be one sentence or phrase

File: pkg/rancher-desktop/assets/specs/command-api.yaml [hunk 2]
```diff
@@ -238,50 +238,63 @@ components:
...
         containerEngine:
           type: object
           properties:
             name:
               type: string
-              enum: ['containerd', 'docker', 'moby']
-              x-rd-usage: Set engine to containerd or moby (aka docker).
-              x-rd-aliases:
-                - "container-engine"
+              # TODO "docker" setting should be a hidden alias of "moby".minimum:
```
Comment (ericpromislow) [inline pkg/rancher-desktop/assets/specs/command-api.yaml:283]: Chris thanked me for getting `docker` in there. We currently have aliases for option names but not values

Comment (jandubois) [reply pkg/rancher-desktop/assets/specs/command-api.yaml:283]: #4113 includes a request to make hidden aliases available.

I'm fine with silently supporting `docker` as an alias for `moby` (or vice versa), but having redundant aliases in descriptions, help texts, error messages is just useless clutter.

**Review by ericpromislow — APPROVED** [review]
LGTM

**Pull request merged**